pkg/resources/allocator.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 Intel Corp. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"sort"
+
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+)
+
+// DeviceSet is to hold and manipulate a set of HostDevice
+type DeviceSet map[string]types.HostDevice
+
+// PackedAllocator implements the Allocator interface
+type PackedAllocator struct {
+}
+
+// Allocator implements the Allocator interface
+type Allocator struct {
+}
+
+// NewPackedAllocator create instance of PackedAllocator
+func NewPackedAllocator() types.Allocator {
```

Comment:
can you return *PackedAllocator here ?
users of this method can then assign this value to types.Allocator typed variable